src/functional/iterator/transforms/unroll_reduce.py

```diff
@@ -16,70 +67,14 @@ class UnrollReduce(NodeTranslator):
     def apply(cls, node: ir.Node, **kwargs):
         return cls().visit(node, **kwargs)
 
-    @staticmethod
-    def _find_connectivities(reduce_args: Iterable[ir.Expr], offset_provider):
-        connectivities = []
-        for arg in reduce_args:
-            if (
-                isinstance(arg, ir.FunCall)
-                and isinstance(arg.fun, ir.FunCall)
-                and arg.fun.fun == ir.SymRef(id="shift")
-            ):
-                assert isinstance(arg.fun.args[-1], ir.OffsetLiteral), f"{arg.fun.args}"
-                connectivities.append(offset_provider[arg.fun.args[-1].value])
-            # if the argument is a lift call by itself look at its arguments recursively
-            elif (
-                isinstance(arg, ir.FunCall)
-                and isinstance(arg.fun, ir.FunCall)
-                and arg.fun.fun == ir.SymRef(id="lift")
-            ):
-                connectivities.extend(UnrollReduce._find_connectivities(arg.args, offset_provider))
```

Comment:
This change was removed by merge conflict, have to think about if that actually did the right thing...

Reply:
I think it didn't make sense.
```
def foo(it):
    return deref(shift(1)(it))

...
def bar(it):
    return reduce(plus, 0)(lift(foo)(shift(offset)(it)))
                                           ^^^^^^
```
it would detect marked offset as the partial shift for reduce which doesn't make sense. I don't see a case where the change introduced in #1055 makes sense...

Reply:
Maybe I see now, let me think again...

Reply:
This is meant for cases where `ìt` is not partially shifted, e.g. just a regular iterator as given to a stencil in a closure. I introduced it because the lift inliner now transforms `shift(...)(lift(...)(...))` (supported by `find_connectivity`) into `lift(...)(shift(...)(...))` (something that has been equivalent before already), which is not supported here.

```
reduce(plus, 0)(lift(deref)(shift("V2V")(it)))
```